### PR TITLE
Added support for new update fields introduced in Bot Api v5.1

### PR DESCRIPTION
**Pull request by ravinderpayal**
Related Issue: https://github.com/bot4s/telegram/issues/96

### CODE REVIEW

File: core/src/com/bot4s/telegram/models/ChatInviteLink.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  * @param inviteLink             The invite link. If the link was created by another chat administrator, then the second
+  *                               part of the link will be replaced with “…”.
+  * @param isPrimary              True, if the link is primary
+  * @param isRevoked              True, if the link is revoked
+  * @param memberLimit            Optional. Maximum number of users that can be members of the chat simultaneously after joining the chat
+  *                               via this invite link; 1-99999
+  */
+case class  ChatInviteLink(
+               creator               : User,
+               expireDate            : Int,
```
Comment (ex0ns) [inline core/src/com/bot4s/telegram/models/ChatInviteLink.scala:16]: This is an optional field, this should be set to `Option[Int]`

File: core/src/com/bot4s/telegram/models/ChatInviteLink.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+               inviteLink            : String,
+               isPrimary             : Boolean,
+               isRevoked             : Boolean,
+               memberLimit           : ChatMember
```
Comment (ex0ns) [inline core/src/com/bot4s/telegram/models/ChatInviteLink.scala:20]: Same here, all optional field should be represented as `Option`

**Review by ex0ns — COMMENTED** [review]
This is looking good ! Thank you
Please see attached comments about options.

It also seems that you have an issue with formatting, please make sure to correctly format the code.

File: core/src/com/bot4s/telegram/models/ChatInviteLink.scala (hunk 2 above)
Comment (ravinderpayal) [reply core/src/com/bot4s/telegram/models/ChatInviteLink.scala:20]: ohh  yeah in hurry, didn't correct them in case class, let me do it right away

Comment (ex0ns) [reply core/src/com/bot4s/telegram/models/ChatInviteLink.scala]: You can also add the default `= None`

Comment (ravinderpayal) [reply core/src/com/bot4s/telegram/models/ChatInviteLink.scala]: See now

**Comment by ravinderpayal** [comment]
Made corrections

**Comment by ex0ns** [comment]
The case classes look good, but you need to also add a Decoder and an Encoder for Circe to work https://github.com/bot4s/telegram/blob/master/core/src/com/bot4s/telegram/marshalling/CirceEncoders.scala and https://github.com/bot4s/telegram/blob/master/core/src/com/bot4s/telegram/marshalling/CirceDecoders.scala

This is need only for the new classes (`ChatInviteLink` and `ChatMemberUpdated`)

**Comment by ravinderpayal** [comment]
I don't have working experience with Circle, but based on what I saw inside files we only need to add 2 lines in each file like this?

implicit val chatInviteLinkEncoder: Encoder[ChatInviteLink] = deriveConfiguredEncoder[ChatInviteLink]
implicit val chatMemberUpdatedEncoder: Encoder[ChatMemberUpdated] = deriveConfiguredEncoder[ChatMemberUpdated]
  

and 

  implicit val chatInviteLinkDecoder: Decoder[ChatInviteLink] = deriveDecoder[ChatInviteLink]
  implicit val chatMemberUpdatedDecoder: Decoder[ChatMemberUpdated] = deriveDecoder[ChatMemberUpdated]

**Comment by ex0ns** [comment]
Yes exactly !

**Comment by eadventurous** [comment]
Are there any plans to merge it? If you need any help feel free to tag me. We are also waiting for this to be merged for the https://github.com/soramitsu/nakayoshi bot fix.

**Comment by ex0ns** [comment]
I'm waiting for the latest changed from @ravinderpayal 
I'll take over this pull request in case it does not change before the end of the week

**Comment by ravinderpayal** [comment]
Hi @ex0ns  let me do it right away.

**Comment by ravinderpayal** [comment]
@ex0ns please look into this

**Comment by ex0ns** [comment]
This is perfect thanks ! I was able to run a random bot and add it to a group without it crashing

**Comment by eadventurous** [comment]
Thanks! Will there be a new version published soon or is it better to directly depend on GitHub `master` for a while?

**Comment by ex0ns** [comment]
I hate to say that but I think you're better off using the version on master for the moment.
I'm still in the process of updating this repo and the publishing on releases/snapshot is not ready yet, it's in my todo list with high priority but I don't have a timeline yet.

**Comment by eadventurous** [comment]
Ok, thanks for the info. We will be looking forward to the new version being published, but we will use the master branch for now then.

**Comment by eadventurous** [comment]
@ex0ns is it possible to reach you by email or in some chat? I am struggling with this dependency update, and would be glad if you have time to answer a couple of questions.

**Comment by eadventurous** [comment]
Seems like the telegram community group of this project is abandoned.

**Comment by ravinderpayal** [comment]
@eadventurous  feel free to ping me or ask here.

**Comment by ex0ns** [comment]
You should also try to ask it  on the telegram group. I'm try to revive this project and I will try to answer the questions on the telegram group as well.
Please feel free to open Github issue as well (they are, IMO the best way to keep track of issues)

**Pull request merged**